### PR TITLE
Add Catalog Customizations Guide

### PR DESCRIPTION
**Pull request by tubbo**
Add a "how-to" guide that explains the setup of catalog customizations,
as well as the additional pricing charge, since there's nothing that
currently documents this out-of-box solution for customized products.

### CODE REVIEW

**Comment by tubbo** [comment]
Before proceeding further and getting more detailed, I wanted to get some feedback on my approach so far. Did I miss anything? Am I wrong about something? Let me know!

File: docs/source/articles/add-catalog-customizations.html.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: Add Catalog Customizations
+excerpt: This guide shows you how to implement user-defined customizations for the products in your catalog.
+---
+
+# Add Catalog Customizations
+
+Customizations are pre-defined fields in your product that users can edit and create custom versions of your projects, such as for custom engravings.
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md]:
```suggestion
Customizations are pre-defined fields for your product that customers can edit and create custom versions. Some examples of how these have been used include:
* Custom engravings for jewelry
* Monograms for shirts
* Email addresses for digital gift cards 
```

File: docs/source/articles/add-catalog-customizations.html.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Step 1: Create a `Catalog::Customizations` subclass
+
+The first step is to subclass `Workarea::Catalog::Customizations` with something descriptive of your customization. In this case, we'll be using the concept of an "engraving" for jewelry or other metal products:
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md]:
```suggestion
The first step is to subclass `Workarea::Catalog::Customizations` with something descriptive of your customization. In this case, we'll be using the concept of an "engraving" for jewelry:
```

File: docs/source/articles/add-catalog-customizations.html.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+```ruby
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md]:
```suggestion
```ruby
# app/models/workarea/catalog/customizations/engraving.rb
```

File: docs/source/articles/add-catalog-customizations.html.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+module Workarea
+  module Catalog
+    class Customizations::Engraving < Customizations
+      customized_fields :initials
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md]:
```suggestion
      customized_fields :initials # declare the field names to be customized
```

File: docs/source/articles/add-catalog-customizations.html.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      validates_presence_of :initials
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md]:
```suggestion
      validates_presence_of :initials # this field is required
```

File: docs/source/articles/add-catalog-customizations.html.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    end
+  end
+end
+```
+
+This validates the `:initials` field when products that use this customization are ordered.
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md]:
```suggestion
You can use any of the standard Rails `ActiveModel` validations in your `Customizations` subclass.
```

File: docs/source/articles/add-catalog-customizations.html.md (hunk 5 above)
Comment (tubbo) [reply docs/source/articles/add-catalog-customizations.html.md:23]: i mean, kinda thought "validates presence of initials" spelled it out for ya?

Comment (bencrouse) [reply docs/source/articles/add-catalog-customizations.html.md:23]: not sure what I was thinking... 🥇

File: docs/source/articles/add-catalog-customizations.html.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+something descriptive of your customization. In this case, we'll be using the
+concept of an "engraving" for jewelry. Create a file named
+**app/models/workarea/catalog/customizations/engraving.rb** and populate it
+with the following class definition:
+
+```ruby
+module Workarea
+  module Catalog
+    class Customizations::Engraving < Customizations
+      customized_fields :initials
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md:28]: Can you add some inline explanation of what you'd want to put here and why?

File: docs/source/articles/add-catalog-customizations.html.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+`:customizations` field on the model:
+
+```ruby
+product = Workarea::Catalog::Product.find('0-XABC12345')
+product.update!(customizations: 'engraving')
+```
+
+(You can also do this in the admin)
+
+Now, any time someone orders this product, they will have the option to engrave
```
Comment (bencrouse) [inline docs/source/articles/add-catalog-customizations.html.md:66]:
```suggestion
Now, any time someone adds this product to their cart, they will have the option to engrave
```

File: docs/source/articles/add-catalog-customizations.html.md (hunk 7 above)
Comment (tubbo) [reply docs/source/articles/add-catalog-customizations.html.md:28]: oh sorry i thought that was grouped with your other change, i'll commit it.

**Comment by tubbo** [comment]
lol thank you `git rebase -i` for getting me out of a jam!

File: docs/source/articles/add-catalog-customizations.html.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+```ruby
+order = Workarea::Order.last
+order.items.first.customizations
+# => { initials: "ABC" }
+```
+
+This would then get passed on to your ERP or OMS for further processing as you
+see fit.
+
+## Step 4: Add Customizations to Product Template(s)
```
Comment (tubbo) [inline docs/source/articles/add-catalog-customizations.html.md:80]: here's the product template stuff. looking at the cart controller i don't think we need to make any decorations there, customizations should be handled along with the rest of the item attrs.

**Pull request merged**